ruby/ql/lib/codeql/ruby/security/UrlRedirectCustomizations.qll

```diff
@@ -121,6 +132,27 @@ module UrlRedirect {
    */
   class StringInterpolationAsSanitizer extends PrefixedStringInterpolation, Sanitizer { }
 
+  /**
+   * A sanitizer for the hostname of a URL.
+   */
+  class HostnameSanitizer extends Sanitizer {
+    HostnameSanitizer() { this = DataFlow::BarrierGuard<hostnameGuard/3>::getABarrierNode() }
+  }
+
+  /**
+   * An in-sanitizer for the hostname of a URL.
+   */
+  class HostnameSanitizerIn extends SanitizerIn {
+    HostnameSanitizerIn() { hostnameSanitizingPrefixEdge(_, this) }
+  }
```

Comment:
It is not necessary to add both in- and out-barriers in order to block flow along this edge, and in fact the in-barrier will block too much flow.

For example,
```rb
"https://#{host}/#{path}"
```
In this case, we want to block flow from `path` but not from `host`. The in-barrier you've added here, however, ends up blocking all flow into the string expression.